src/lib_json/json_value.cpp

```diff
@@ -1155,10 +1155,14 @@ Value const& Value::operator[](CppTL::ConstString const& key) const {
 }
 #endif
 
-Value& Value::append(const Value& value) { return (*this)[size()] = value; }
-
+Value& Value::append(const Value& value) { return append(Value(value)); }
```

Comment:
Are we going to do parameter checking here, so that we don't have to jump to next function when we encounter an unexpected parameter type？

Reply:
Considering the move version of this method does parameter checking, I'd rather optimize for the good case and not have duplicate parameter checking between multiple methods.